src/handlers/rotate.rs

```diff
@@ -0,0 +1,126 @@
+    if image_directory.is_err() {
+        return Err((StatusCode::NOT_FOUND, "Image not found!".to_owned()));
+    }
+
+    let image_directory = image_directory.unwrap();
+    let image_directory = image_directory.to_str().unwrap();
+
+    let image_path = determine_img_path(image_directory, query.id);
+    if image_path.is_err() {
+        log::warn!(
+            "Image not found where the path was previously determined. Id: {:?}, Directory: {:?}, Error: {:?}",
+            query.id,
+            image_directory,
+            image_path.err()
+        );
+        return Err((StatusCode::NOT_FOUND, "Image not found!".to_owned()));
+    }
+
+    let image_path = image_path.unwrap();
+    let image_path = image_path.to_str().unwrap();
+
+    let raw_image = match std::fs::read(image_path) {
+        Ok(raw_image) => raw_image,
+        Err(err) => {
+            log::error!(
+                "Error while reading image. Id: {:?}, Error: {:?}, Path: {:?}",
+                query.id,
+                err,
+                image_path
+            );
+            return Err((
+                StatusCode::INTERNAL_SERVER_ERROR,
+                "Error while reading image!".to_owned(),
+            ));
+        }
+    };
+
+    // We need to fully load the image into memory here, as saving a VipsImage to a file it was
+    // memory mapped from is a bad idea. The best solution would be to use `vips_image_new_from_file`
+    // and set memory argument to true (to force loading into memory), but that's not possible
+    // with the rust bindings.
+    let image = match VipsImage::new_from_buffer(raw_image.as_slice(), "") {
+        Ok(image) => image,
+        Err(err) => {
+            log::error!(
+                "Error while opening image. Id: {:?}, Error: {:?}, Path: {:?}",
+                query.id,
+                err,
+                image_path
+            );
+            return Err((
+                StatusCode::INTERNAL_SERVER_ERROR,
+                "Error while opening image!".to_owned(),
+            ));
+        }
+    };
+
+    let rotated = match ops::rotate(&image, query.angle) {
+        Ok(rotated) => rotated,
+        Err(err) => {
+            log::error!(
+                "Error while rotating image. Id: {:?}, Error: {:?}",
+                query.id,
+                err
+            );
+            return Err((
+                StatusCode::INTERNAL_SERVER_ERROR,
+                "Error while rotating image!".to_owned(),
+            ));
+        }
+    };
+
+    match save_image(&rotated, image_path) {
+        Ok(_) => (),
+        Err(err) => {
+            log::error!(
+                "Error while saving image. Id: {:?}, Error: {:?}",
+                query.id,
+                err
+            );
+            return Err((
+                StatusCode::INTERNAL_SERVER_ERROR,
+                "Error while saving image!".to_owned(),
+            ));
+        }
+    }
```

Comment:
As discussed, I would prefer the "save as new name and move in place" method

Reply:
Changed.